dataikuapi/dss/webapp.py

```diff
@@ -0,0 +1,121 @@
+import sys
+from .future import DSSFuture
+
+if sys.version_info >= (3,0):
+  import urllib.parse
+  dku_quote_fn = urllib.parse.quote
+else:
+  import urllib
+  dku_quote_fn = urllib.quote
+
+
+class DSSWebApp(object):
+    """
+    A handle to manage a webapp
+    """
+    def __init__(self, client, project_key, webapp_id):
+        """Do not call directly, use :meth:`dataikuapi.dss.project.DSSProject.get_webapps`"""
+        self.client = client
+        self.project_key = project_key
+        self.webapp_id = webapp_id
+
+    def get_state(self):
+        """
+        Return the state of the webapp
+
+        :return: the state of the webapp
+        :rtype: :class:`DSSWebAppBackendState`
+        """
+        state = self.client._perform_json("GET", "/projects/%s/webapps/%s/backend/state" % (self.project_key, self.webapp_id))
+        return DSSWebAppBackendState(self.client, self.project_key, self.webapp_id, state)
+
+    def stop_backend(self):
+        """
+        Stop a webapp
+        """
+        self.client._perform_empty("PUT", "/projects/%s/webapps/%s/backend/actions/stop" % (self.project_key, self.webapp_id))
+
+    def restart_backend(self):
+        """
+        Restart a webapp
+        :returns: a handle to a DSS future to track the progress of the restart
+        :rtype: :class:`dataikuapi.dss.future.DSSFuture`
+        """
+        future = self.client._perform_json("PUT", "/projects/%s/webapps/%s/backend/actions/restart" % (self.project_key, self.webapp_id))
+        return DSSFuture(self.client, future["jobId"])
+
+    def get_definition(self):
```

Comment:
"get_definition" is the legacy naming for legacy methods that return a dict. Modern methods that return settings must be called get_settings and return an object called XXXSettings